File: app/src/main/java/com/example/chi_12_di/domain/usecase/impl/DeleteAllPhotosUseCaseImplModule.java
package com.example.chi_12_di.domain.usecase.impl;

import com.example.chi_12_di.domain.usecase.api.IDeleteAllPhotosUseCase;

import dagger.Module;
import dagger.Provides;

@Module
public class DeleteAllPhotosUseCaseImplModule {

    @Provides
    IDeleteAllPhotosUseCase provideIDeleteAllPhotosUseCase (DeleteAllPhotosUseCaseImpl deleteAllPhotosUseCaseImpl){
        return deleteAllPhotosUseCaseImpl;
    }
}

File: app/src/main/java/com/example/chi_12_di/di/DeleteAllPhotosUseCaseImplModule.kt
package com.example.chi_12_di.di

import com.example.chi_12_di.domain.usecase.api.IDeleteAllPhotosUseCase
import com.example.chi_12_di.domain.usecase.impl.DeleteAllPhotosUseCaseImpl
import dagger.Binds
import dagger.Module

@Module
interface DeleteAllPhotosUseCaseImplModule {

    @Binds
    fun provideIDeleteAllPhotosUseCase(deleteAllPhotosUseCaseImpl: DeleteAllPhotosUseCaseImpl): IDeleteAllPhotosUseCase
}
